fotos/delete.ts

```diff
@@ -1,65 +1,217 @@
+      throw new JSONParseError(e, "invokeGetImageRecord");
+    });
+
+}
+
+export function deleteObject(s3, s3Params) {
+  return s3.deleteObject(s3Params).promise()
+    .catch((e) => {
+      throw new DeleteObjectError(e, s3Params.Key, s3Params.Bucket);
+    });
+}
+
+export function deleteImageRecord(ddbParams) {
+  return dynamodb.delete(ddbParams).promise()
+    .catch((e) => {
+      throw new DeleteRecordError(e, ddbParams);
+    });
+}
+export function getFaceIds(image: IImage): string[] {
+  return image && image.faces ?
+    image.faces
+      .filter((f) => f.Face!.FaceId !== undefined)
+      .map((f) => f.Face!.FaceId!) :
+    [];
+}
+
+export function deleteFacesInImage(image: IImage): Promise<DeleteFacesResponse> | DeleteFacesResponse {
+  const faces = getFaceIds(image);
+  if (faces.length > 0) {
+    const rekognitionClient = new Rekognition();
+    const params: DeleteFacesRequest = {
+      CollectionId: process.env.FOTOPIA_GROUP!,
+      FaceIds: faces,
+    };
+    return rekognitionClient.deleteFaces(params).promise();
+  } else {
+    return {
+      DeletedFaces: [],
+    };
+  }
+}
+
+export function getInvokeQueryParams(image: IImage, loggerBaseParams) {
+  const request: IQueryBody = {
+    criteria: {
+      people: image.people!,
+      tags: [],
+    },
+    from: 0,
+    to: Date.now(),
+  };
   return {
-    FunctionName: process.env.IS_OFFLINE ? "get" : `${process.env.LAMBDA_PREFIX}get`,
+    FunctionName: process.env.IS_OFFLINE ? "query" : `${process.env.LAMBDA_PREFIX}query`,
     InvocationType: INVOCATION_REQUEST_RESPONSE,
     LogType: "Tail",
     Payload: JSON.stringify({
-      pathParameters: {
+      body: JSON.stringify({
         ...request,
+        traceMeta: getTraceMeta(loggerBaseParams),
+      }),
+    }),
+  };
+}
+
+export function getPeopleWithImages(image: IImage, queriedImages: IImage[]): IPersonWithImages[] {
+  return image.people!.map((personId): IPersonWithImages => ({
+    id: personId,
+    imageIds: queriedImages.filter((qImg) => qImg.people!.includes(personId)).map((qImg) => qImg.id),
+  }));
+}
+
+export function queryImagesByPeople(image: IImage, loggerBaseParams): Promise<IPersonWithImages[]> {
+  const params = getInvokeQueryParams(image, loggerBaseParams);
+  return lambda.invoke(params).promise()
+  .then((invocationResponse: InvocationResponse) => parseQueryResponse(invocationResponse, image));
+}
+
+export function parseQueryResponse(invocationResponse: InvocationResponse, image: IImage) {
+  try {
+    const payload = JSON.parse(invocationResponse.Payload as string);
+    const queriedImages: IImage[] = JSON.parse(payload.body);
+    return image && image.people && Array.isArray(queriedImages) ?
+      getPeopleWithImages(image, queriedImages) :
+      [];
+  } catch (e) {
+    throw new JSONParseError(e, `parseQueryResponse: ${JSON.stringify(invocationResponse)}`);
+  }
+}
+
+export function getDeletePeople(peopleImages: IPersonWithImages[]): string[] {
```

Comment:
Delete - delete unique people (not deleting face ids, eg that are matched to other people... may be an issue)